test/common/protobuf/utility_test.cc

```diff
@@ -261,6 +263,364 @@ TEST_F(ProtobufUtilityTest, LoadTextProtoFromFile_Failure) {
+    - '[redacted]'
+    - '[redacted]'
+sensitive_repeated_message:
+  - insensitive_string: '[redacted]'
+    insensitive_repeated_string:
+      - '[redacted]'
+      - '[redacted]'
+  - insensitive_string: '[redacted]'
+    insensitive_repeated_string:
+      - '[redacted]'
+      - '[redacted]'
+sensitive_any:
+  '@type': type.googleapis.com/envoy.test.Sensitive
+  insensitive_string: '[redacted]'
+  insensitive_repeated_string:
+    - '[redacted]'
+    - '[redacted]'
+sensitive_repeated_any:
+  - '@type': type.googleapis.com/envoy.test.Sensitive
+    insensitive_string: '[redacted]'
+    insensitive_repeated_string:
+      - '[redacted]'
+      - '[redacted]'
+  - '@type': type.googleapis.com/envoy.test.Sensitive
+    insensitive_string: '[redacted]'
+    insensitive_repeated_string:
+      - '[redacted]'
+      - '[redacted]'
+sensitive_typed_struct:
+  type_url: type.googleapis.com/envoy.test.Sensitive
+  value:
+    insensitive_string: '[redacted]'
+    insensitive_repeated_string:
+      - '[redacted]'
+      - '[redacted]'
+sensitive_repeated_typed_struct:
+  - type_url: type.googleapis.com/envoy.test.Sensitive
+    value:
+      insensitive_string: '[redacted]'
+      insensitive_repeated_string:
+        - '[redacted]'
+        - '[redacted]'
+  - type_url: type.googleapis.com/envoy.test.Sensitive
+    value:
+      insensitive_string: '[redacted]'
+      insensitive_repeated_string:
+        - '[redacted]'
+        - '[redacted]'
+sensitive_v2_data_source:
+  - {}
+  - filename: This filename should not be redacted despite its parent.
+  - inline_string: '[redacted]'
+  - inline_string: '[redacted]'
+sensitive_v3alpha_data_source:
+  - {}
+  - filename: This filename should not be redacted despite its parent.
+  - inline_string: '[redacted]'
+  - inline_string: '[redacted]'
+insensitive_any:
+  '@type': type.googleapis.com/envoy.test.Sensitive
+  sensitive_string: '[redacted]'
+  sensitive_repeated_string:
+    - '[redacted]'
+    - '[redacted]'
+insensitive_repeated_any:
+  - '@type': type.googleapis.com/envoy.test.Sensitive
+    sensitive_string: '[redacted]'
+    sensitive_repeated_string:
+      - '[redacted]'
+      - '[redacted]'
+  - '@type': type.googleapis.com/envoy.test.Sensitive
+    sensitive_string: '[redacted]'
+    sensitive_repeated_string:
+      - '[redacted]'
+      - '[redacted]'
+insensitive_typed_struct:
+  type_url: type.googleapis.com/envoy.test.Sensitive
+  value:
+    sensitive_string: '[redacted]'
+    sensitive_repeated_string:
+      - '[redacted]'
+      - '[redacted]'
+insensitive_repeated_typed_struct:
+  - type_url: type.googleapis.com/envoy.test.Sensitive
+    value:
+      sensitive_string: '[redacted]'
+      sensitive_repeated_string:
+        - '[redacted]'
+        - '[redacted]'
+  - type_url: type.googleapis.com/envoy.test.Sensitive
+    value:
+      sensitive_string: '[redacted]'
+      sensitive_repeated_string:
+        - '[redacted]'
+        - '[redacted]'
```

Comment:
WDYT of exposing a different between redacted & non-empty vs redacted & empty?

Reply:
Hmm... I guess that would be to support the case where just knowing that a field was set is, in itself, sensitive. My guess is that we don't care about that, but I'll defer to @PiotrSikora and @incfly - what do you think?

Reply:
Yeah that might be a granularity worth considering. My use-case was that I thought an admin user might want to know whether a cert was set at all.

My original proposal was to hash the cert so you could tell at a glance if you are looking at two, whether they are the same. But I could see the argument that the hash of a cert is sensitive. Also interested in Piotr's take on that.

Reply:
There is a huge value of knowing whether something was set or not when debugging, so I'd prefer to see the `[redacted]` value only for the configured fields, and leave it empty for the unset fields, instead of always having `[redacted]` in the output, even if this leaks some information (though, I wouldn't call it sensitive).

Reply:
Ok cool, that's what is currently implemented, thanks!